### PR TITLE
Convert community care appointment fetches to VAOS service

### PR DESCRIPTION
**Pull request by LeKeve**
## Description

This PR updates cc appointment fetches to the v2 VAOS services.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#27332

## Testing done
- local, unit testing

## Screenshots
N/A

## Acceptance criteria
- [x] The services listed in current services are not called when VAOS service cc appointments feature toggle is on
- [x] The services mentioned as replacement services are called when VAOS service cc appointments feature toggle is on
- [x] Data from the replacement service should be transformed in a form submit transformer to match the VAOS service expected format

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/services/appointment/index.unit.spec.js [hunk 1]
```diff
@@ -73,11 +73,7 @@ describe('VAOS Appointment service', () => {
           startDate,
         ).toISOString()}&end_date=${moment(endDate).toISOString()}&type=va`,
       );
-      expect(global.fetch.secondCall.args[0]).to.contain(
-        `/vaos/v0/appointments?start_date=${moment(
-          startDate,
-        ).toISOString()}&end_date=${moment(endDate).toISOString()}&type=cc`,
-      );
+      expect(global.fetch.secondCall).to.be.null;
```
Comment (LeKeve) [inline src/applications/vaos/tests/services/appointment/index.unit.spec.js:76]: I suppose I could just get rid of this, but now that the appointment fetches are occurring separately this second cc call doesn't occur on error.

File: src/applications/vaos/services/appointment/index.js [hunk 2]
```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
...
   try {
-    if (useV2) {
-      const appointments = await getAppointments(startDate, endDate, [
-        'booked',
-        'cancelled',
-      ]);
-
-      const appointmentsWithoutRequests = appointments.filter(
-        appt => !appt.requestedPeriods,
+    let appointments = [];
```
Comment (LeKeve) [inline src/applications/vaos/services/appointment/index.js:126]: This logic is a little weird with the both feature flags so I reorganized it a bit.

File: src/applications/vaos/services/appointment/index.js [hunk 3]
```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
...
 
-    const appointments = await Promise.all([
-      getConfirmedAppointments(
-        'va',
-        moment(startDate).toISOString(),
-        moment(endDate).toISOString(),
-      ),
-      getConfirmedAppointments(
+    if (!useV2CC) {
+      const confirmedCCAppointments = await getConfirmedAppointments(
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/index.js:148]: The awaits here change this function from making the two calls in parallel to making them serially. It would be nice to preserve the parallel fetching if we can without the code looking too confusing.

File: src/applications/vaos/services/appointment/index.js [hunk 4]
```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
...
-      return transformVAOSAppointments(appointmentsWithoutRequests);
+      // We might get partial results back from MAS, so throw an error if we do
+      if (hasPartialResults(confirmedVAAppointments)) {
+        throw mapToFHIRErrors(
+          appointments[0].errors,
+          'MAS returned partial results',
+        );
+      }
+
+      appointments = [
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/index.js:142]: If you want to avoid reassigning each time, you could do `appointments.push(...transformConfirmedAppointments(confirmedVAAppointments.data))`

File: src/applications/vaos/services/appointment/index.js [hunk 5]
```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
...
+    if (useV2VA || useV2CC) {
+      const kindsOfVAAppointments = ['clinic', 'telehealth', 'phone'];
+      const allAppointments = await getAppointments(startDate, endDate, [
+        'booked',
+        'cancelled',
+      ]);
+
+      const filteredAppointments = allAppointments.filter(appt => {
+        if (
+          (!useV2VA && kindsOfVAAppointments.includes(appt.kind)) ||
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/index.js:168]: Would it work make this check `&& appt.kind !== 'cc'`?

File: src/applications/vaos/tests/appointment-list/components/CommunityCareAppointmentDetailsPage.unit.spec.jsx [hunk 6]
```diff
@@ -563,3 +569,215 @@ describe('VAOS <CommunityCareAppointmentDetailsPage>', () => {
...
+    const url = '/cc/01aa456cc';
+    const appointmentTime = moment().add(1, 'days');
+    const start = moment()
+      .subtract(30, 'days')
+      .format('YYYY-MM-DD');
+    const end = moment()
+      .add(395, 'days')
+      .format('YYYY-MM-DD');
+
+    const appointment = getVAOSAppointmentMock();
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/CommunityCareAppointmentDetailsPage.unit.spec.jsx:592]: Not at all required, but you could try out the new createAppointmentMockByVersion function I created, since that's merged now.

File: src/applications/vaos/tests/appointment-list/components/UpcomingAppointmentsList.unit.spec.jsx [hunk 7]
```diff
@@ -600,6 +600,7 @@ describe('VAOS <UpcomingAppointmentsList> V2 api', () => {
       featureToggles: {
         ...initialState.featureToggles,
         vaOnlineSchedulingVAOSServiceVAAppointments: true,
+        vaOnlineSchedulingVAOSServiceCCAppointments: true,
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/UpcomingAppointmentsList.unit.spec.jsx:603]: Why was this necessary? I think these are the tests for the non-homepage refresh pages.

File: src/applications/vaos/services/appointment/index.js [hunk 8]
```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
 export async function getBookedAppointments({
   startDate,
   endDate,
-  useV2 = false,
+  useV2VA = false,
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/index.js:122]: We should update the JSDoc comment, too

File: src/applications/vaos/services/appointment/index.js (hunk 5 above)
Comment (LeKeve) [reply src/applications/vaos/services/appointment/index.js:168]: Good call, kind of overcomplicated this bit by creating the va kinds array.

File: src/applications/vaos/appointment-list/redux/actions.js [hunk 9]
```diff
@@ -511,10 +523,21 @@ export function fetchConfirmedAppointmentDetails(id, type) {
         appointment = await fetchBookedAppointment({
           id,
           type,
-          useV2: featureVAOSServiceAppointments,
+          useV2:
```
[the remark this answers is not in the document]
Comment (LeKeve) [reply src/applications/vaos/appointment-list/redux/actions.js:526]: Good catch. Passing by `type` should work, will update.

File: src/applications/vaos/tests/appointment-list/components/UpcomingAppointmentsList.unit.spec.jsx (hunk 7 above)
Comment (LeKeve) [reply src/applications/vaos/tests/appointment-list/components/UpcomingAppointmentsList.unit.spec.jsx:603]: Not necessary. Removing these lines.

Comment (LeKeve) [reply src/applications/vaos/tests/appointment-list/components/UpcomingAppointmentsList.unit.spec.jsx:603]: Actually, I added these because not having them meant getting an error when we fetched cc appointments from the v0 endpoint. So these flags were added so the tests in this block just had to fetch from v2.

File: src/applications/vaos/tests/services/appointment/index.unit.spec.js (hunk 1 above)
Comment (LeKeve) [reply src/applications/vaos/tests/services/appointment/index.unit.spec.js]: Reverted now that both calls happen when both vaos service flags are false.

File: src/applications/vaos/services/appointment/transformers.v2.js [hunk 10]
```diff
@@ -184,6 +183,7 @@ export function transformVAOSAppointment(appt) {
     comment: appt.comment,
     videoData,
     communityCareProvider: null,
+    practitioners: appt.practitioners,
```
Comment (LeKeve) [inline src/applications/vaos/services/appointment/transformers.v2.js:186]: Added this back for confirmed cc appointments and updated the diff tests.

File: src/applications/vaos/tests/appointment-list/components/UpcomingAppointmentsList.unit.spec.jsx (hunk 7 above)
Comment (jbalboni) [reply src/applications/vaos/tests/appointment-list/components/UpcomingAppointmentsList.unit.spec.jsx:603]: Gotcha, I was also wrong about which tests these were.

**Review by jbalboni — APPROVED** [review]
